agent/consul/helper_test.go

```diff
@@ -1213,10 +1213,12 @@ func registerTestRoutingConfigTopologyEntries(t *testing.T, codec rpc.ClientCode
 func registerLocalAndRemoteServicesVIPEnabled(t *testing.T, state *state.Store) {
 	t.Helper()
 
-	_, entry, err := state.SystemMetadataGet(nil, structs.SystemMetadataVirtualIPsEnabled)
-	require.NoError(t, err)
-	require.NotNil(t, entry)
-	require.Equal(t, "true", entry.Value)
+	retry.Run(t, func(r *retry.R) {
```

Comment:
This is aligned with an existing test for Virtual IPs: https://github.com/hashicorp/consul/blob/2e4cb6f77d2be36b02e9be0b289b24e5b0afb794/agent/consul/leader_test.go#L2201